### PR TITLE
#685: updated bazel

### LINKED ISSUE #685: Update Bazel
## Background

Currently, we use Bazel `5.2.0`; however version `7.1.1` is already available. 

## Acceptance Criteria

Use Latest version of Bazel in all flavors.

## Needed changes

- [x]  Use new `MODULE.bazel` instead of `WORKSPACE`
    - [x] Convert the `base` part in `Script-Languages` to a submodule, which then can be used from the `exaudfclient` in `Script-Languages` and `Script-Languages-Release`. This simplifies the packaging of the UDF client in the Script-Languages-Container as we don't need to mess around with symlinks for the binaries in the former `Script-Languages` Bazel package.
    - [x] Convert the rules for external packages (`zmq`, `protobuf`, `Python`, `Java`,...) to be complient to the new `MODULE` implementation
    - [x] Use `MODULE` dependencies were applicable
        - Maven
- [x] Refactor the `Java` BUILD file
    - [x] The new version of Bazel does not support directories in the `out` attribute of a rule. We need to specify each single output file
    - [x] Removed unnecessary rules
    - [x] Pass only needed files to the filter rules
- [x] Clean up Conda flavors
   - [x]  Pin only given packages, but not all (already installed) packages 
   - [x] Remove transitive dependencies in conda package files
   - [x] Test if CUDA flavors still work correctly with `Pytorch` and `Tensorflow` + Nvidia GPU
   - [x] Remove dependency to `OpenJDK`: Bazel defines this dependency. We don't need it explicitely.

### PR DESCRIPTION
**Pull request by tomuben**
fixes #685 

### All Submissions:

* [x] Check if your pull request goes to the correct bash branch (develop or master)?
* [x] Is the title of the Pull Request correct?
* [x] Is the title of the corresponding issue correct?
* [x] Have you updated the changelog?
* [x] Have you checked to ensure there aren't other open [Pull Requests](../../../../pulls) for the same update/change?  
* [x] Are you mentioning the issue which this PullRequest fixes ("Fixes...")

### When integrating to Develop branch:

1. [ ] Remember to merge with "Squash and Merge"

### When integrating to Main branch:

1. [ ] Remember to merge with "Merge"

### CODE REVIEW

File: exaudfclient/BUILD [hunk 1]
```diff
@@ -1,81 +1,108 @@
...
         "//conditions:default": []
     })
 
-R_VM_ENABLED_DEFINE=select({
-        "//:r": ["ENABLE_R_VM"],
-        "//conditions:default": []
-    })
-VM_ENABLED_DEFINES=VM_ENABLED_DEFINES+R_VM_ENABLED_DEFINE
 
 cc_binary(
```
Comment (tkilias) [inline exaudfclient/BUILD:88]: Do we actually still need both exaudfclient_bin and exaudfclient_bin_py3

Comment (tomuben) [reply exaudfclient/BUILD:88]: I guess we don't. I will create a ticket for next Kehrwoche...

Comment (tomuben) [reply exaudfclient/BUILD:88]: See https://github.com/exasol/script-languages-release/issues/944

File: flavors/template-Exasol-all-java-17/flavor_base/build_run/Dockerfile [hunk 2]
```diff
@@ -24,11 +24,9 @@ RUN ldconfig
 RUN mkdir /exaudfclient /exaudf /exaudf/src
 COPY /exaudfclient /exaudfclient
 
-WORKDIR /exaudfclient/extension
+WORKDIR /exaudfclient/
 RUN ["/bin/bash", "-c", "source /env && bash build.sh --config no-tty --config optimize --config java --config fast-binary-py3"]
-RUN cp -r -L bazel-bin/* /exaudf && rm -r /exaudf/external/bazel_tools/
```
Comment (tomuben) [inline flavors/template-Exasol-all-java-17/flavor_base/build_run/Dockerfile:29]: Need to be checked.....

File: doc/changes/changes_8.3.0.md [hunk 3]
```diff
@@ -13,3 +13,4 @@ Code name:
 ## Refactorings
 
 * #933: Added file `release_config.yml`
+* #685: Updated Bazel
```
Comment (tkilias) [inline doc/changes/changes_8.3.0.md:16]:
```suggestion
* #685: Updated Bazel and removed freeze-install from the conda flavors
```

**Review by tkilias — APPROVED** [review]

**Pull request merged**